engine/lib/phx/src/system/bytes.rs

```diff
@@ -1,365 +1,220 @@
-    let value: f64 =
-        *((&mut this.data as *mut libc::c_char).offset(this.cursor as isize) as *mut f64);
-    this.cursor = this
-        .cursor
-        .wrapping_add(std::mem::size_of::<f64>() as libc::c_ulong as u32);
-    value
-}
+    pub fn write_u64(&mut self, value: u64) {
+        let _ = self.cursor.write_u64::<LittleEndian>(value);
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn Bytes_WriteU64(this: &mut Bytes, value: u64) {
-    *((&mut this.data as *mut libc::c_char).offset(this.cursor as isize) as *mut u64) = value;
-    this.cursor = this
-        .cursor
-        .wrapping_add(std::mem::size_of::<u64>() as libc::c_ulong as u32);
-}
+    pub fn write_i8(&mut self, value: i8) {
+        let _ = self.cursor.write_i8(value);
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn Bytes_ReadI16(this: &mut Bytes) -> i16 {
-    let value: i16 =
-        *((&mut this.data as *mut libc::c_char).offset(this.cursor as isize) as *mut i16);
-    this.cursor = this
-        .cursor
-        .wrapping_add(std::mem::size_of::<i16>() as libc::c_ulong as u32);
-    value
-}
+    pub fn write_i16(&mut self, value: i16) {
+        let _ = self.cursor.write_i16::<LittleEndian>(value);
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn Bytes_ReadI32(this: &mut Bytes) -> i32 {
-    let value: i32 =
-        *((&mut this.data as *mut libc::c_char).offset(this.cursor as isize) as *mut i32);
-    this.cursor = this
-        .cursor
-        .wrapping_add(std::mem::size_of::<i32>() as libc::c_ulong as u32);
-    value
-}
+    pub fn write_i32(&mut self, value: i32) {
+        let _ = self.cursor.write_i32::<LittleEndian>(value);
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn Bytes_WriteF64(this: &mut Bytes, value: f64) {
-    *((&mut this.data as *mut libc::c_char).offset(this.cursor as isize) as *mut f64) = value;
-    this.cursor = this
-        .cursor
-        .wrapping_add(std::mem::size_of::<f64>() as libc::c_ulong as u32);
-}
+    pub fn write_i64(&mut self, value: i64) {
+        let _ = self.cursor.write_i64::<LittleEndian>(value);
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn Bytes_Print(this: &Bytes) {
-    info!("{} bytes:", this.size);
+    pub fn write_f32(&mut self, value: f32) {
+        let _ = self.cursor.write_f32::<LittleEndian>(value);
+    }
 
-    for i in 0..this.size {
-        libc::putchar(*(&this.data as *const libc::c_char).offset(i as isize) as i32);
+    pub fn write_f64(&mut self, value: f64) {
+        let _ = self.cursor.write_f64::<LittleEndian>(value);
     }
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn Bytes_Save(this: &Bytes, path: *const libc::c_char) {
-    let mut file = File_Create(path).unwrap_or_else(|| unsafe {
-        panic!(
-            "Bytes_Save: Failed to open file '{:?}' for writing",
-            CStr::from_ptr(path)
-        )
-    });
-    let _ = file.file.write_all(this.as_slice());
+pub extern "C" fn Bytes_GetData(this: &mut Bytes) -> *mut libc::c_void {
```

Comment:
Why this is still old style?

Reply:
It was still used in a few places but I didn't want to expose it to Lua (because of the raw pointer). However, I noticed that I've removed all the uses now, so this can go.